extensions/amp-pinterest/0.1/pin-widget.js

```diff
@@ -336,4 +342,17 @@ export class PinWidget {
 
     return this.fetchPin().then(this.renderPin.bind(this));
   }
+
+  /**
+   * Determine the height of the contents to allow resizing after first layout.
+   *
+   * @return {number|null}
+   */
+  height() {
+    return (
+      this.heightOwnerElement_
+        .getBoundingClientRect /* REVIEW */
```

Comment:
Should this height calculation be wrapped in a `vsync` mutator?

Reply:
If possible I think we should not use `getBoundingClientRect` after all.

A single use InObs seems to be the best way to get this measurement without reflow/recalc.

```js
function getIntersectionEntryAsync(element) {
  return new Promise(resolve => {
    const inobs = new IntersectionObserver(entries => {
      inobs.disconect():
      resolve(entries[entries.length - 1]);
    });
    inobs.observe(element);
  });
}
```

I think @samouri introduced a util to the same effect recently.

Reply:
Yep! It is called `measureIntersection`. 

https://github.com/ampproject/amphtml/blob/4da8b96882786565d2ae2d2725ed1c6ecb9fb300/src/utils/intersection.js#L58-L67

To get the height you'd call:
```js
measureIntersection(element).then(entry => entry.boundingClientRect.height);
```

---

Although TBH I'm not sure which is preferably right now. Using that or `measurePageLayoutBox` which does a measure in a vsync

https://github.com/ampproject/amphtml/blob/4da8b96882786565d2ae2d2725ed1c6ecb9fb300/src/utils/page-layout-box.js#L37-L44